orderBook.js

```diff
@@ -0,0 +1,51 @@
+function isOrderMatch (incomingOrder, existingOrder)
+{
+  if (incomingOrder.price === existingOrder.price &&
+    incomingOrder.type !== existingOrder.type)
+  {
+    return true
+  }
+  return false
+}
+
+function handleOrder (book, incomingOrder, index)
+{
+  if (incomingOrder.quantity === book[ index ].quantity)
+  {
+    book.splice(index, 1)
+
+    return book
+  } else if (incomingOrder.quantity > book[ index ].quantity)
+  {
+    incomingOrder.quantity -= book[ index ].quantity
+    book.splice(index, 1)
+
+    return reconcileOrder(book, incomingOrder)
+  } else
+  {
+    book[ index ].quantity -= incomingOrder.quantity
+    let tempOrder = book.splice(index, 1)
+    book.push(tempOrder[0])
+
+    return book
+  }
```

Comment:
This function is really doing two things. It is reconciling a pair of orders but it is also altering the structure of the book. You should instead have a function that takes two orders, reconciles them, and returns what remains to reconcileOrder which can then call a different function for altering the book.